internal/bundle/plain_test.go

```diff
@@ -0,0 +1,96 @@
+package bundle
+
+import (
+	"io/fs"
+	"testing/fstest"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Plain bundle v0 loader", func() {
+	It("should fail if is no manifests directory", func() {
+		mockfs := fstest.MapFS{}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("file does not exist"))
+	})
+	It("should fail if there is an empty manifests directory", func() {
+		mockfs := fstest.MapFS{
+			"manifests": {
+				Mode: fs.ModeDir,
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("found zero files"))
+	})
+	It("should fail if there is a file, but no yaml objects", func() {
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte(""),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("required to contain at least one object"))
+	})
+	It("should fail if the manifests directory contains a file with invalid json or yaml", func() {
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte("invalid;yaml"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("cannot unmarshal string into Go value"))
+	})
+	It("should fail if any of the files are invalid json or yaml", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+			"manifests/invalid.yaml": {
+				Data: []byte("not;valid"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("cannot unmarshal string into Go value"))
+	})
+	It("should fail if the YAML or JSON objects do not specify `kind`", func() {
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte("almost: valid"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("Object 'Kind' is missing"))
+	})
+	It("should pass if the JSON/YAML is valid and contains a `kind` element", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err).To(BeNil())
+	})
+
+	It("should fail if there are nested subdirectories", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+			"manifests/invalid/valid.yaml": {
+				Data: []byte("kind: illegalsubdir"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("subdirectories are not allowed"))
+	})
+	It("should pass with multi-object yaml files", func() {
```

Comment:
Can you also add tests for:
1. All manifest files contain JSON
2. Manifest files are a mix of JSON and YAML (but each file is one or the other)
3. The ability to parse and flatten `v1.List` objects that contain a list of `items` that are objects.